Add CVE-2020-14380 to security page and fix indenation (#1674)
theforeman/theforeman.org: security.md

### Disclosure details

#### <a id="2020-14380"></a>CVE-2020-14380: Users can gain elevated rights when logging via external authentication

> Users that are authenticated using `external` authencation methods such as Kerberos, OpenID Connect or Apache can gain administrative rights of an `internal` user. This is possible when the username of an internal and external user is same.

Reviewer: ```suggestion
Users that are authenticated using `external` authentication methods such as Kerberos, OpenID Connect or Apache SSO can gain the rights of an `internal` user. This is possible only when the username of an internal and external user is the same.
```

- Affects external user authentication since Foreman 1.4

> - Fix released in Foreman 2.2.0, 2.1.1 and 2.0.2 and higher

Reviewer: These versions are incorrect, the correct versions are:
```suggestion
- Fix released in Foreman 2.2.0, 2.1.3 and 2.0.3 and higher
```

- Redmine issue [#30739](https://projects.theforeman.org/issues/30739)

> #### <a id="2020-14334"></a>CVE-2020-14334: World readable cache could expose sensitive settigs

Reviewer 2: Unrelated, but looks like there's a typo:
```suggestion
#### <a id="2020-14334"></a>CVE-2020-14334: World readable cache could expose sensitive settings
```
